presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -305,7 +306,7 @@ public ConnectorSplitSource getSplits(
                         table.getDatabaseName(),
                         table.getTableName(),
                         cacheQuotaRequirement,
-                        maxInitialSplits,
+                        getHiveMaxInitialSplitSize(session) != maxInitialSplits ? getHiveMaxInitialSplitSize(session) : maxInitialSplits,
```

Comment:
Why not just `getHiveMaxInitialSplitSize(session)` ? The default value for the session property is the same value from `hiveClientConfig.getMaxInitialSplits()`.

Reply:
+1; same for all other places. This also means we don't need `maxInitialSplits` for `HiveSplitManager`'s constructor

Reply:
Good suggestion, I have remove the `maxInitialSplits` from `HiveSplitManager`'s constructor